### PR TITLE
fix: Correct empty view display

### LINKED ISSUE #222: Empty layout appears at start itself
Currently on any data loading page, empty layout is shown from start itself with progress bar.

Expected: On page start , only progress bar should appear with blank background. Then on data load failure empty layout should appear.

### PR DESCRIPTION
**Pull request by mejariamol**
- Adds showEmptyView method in interface
- Hides empty view on page start
- Hides empty view on swipe refresh
- Modifies empty view layout

fixes #222

img-20170624-wa0002

### CODE REVIEW

**Comment by mejariamol** [comment]
@iamareebjamal @rohanagarwal94 @Shailesh351 @Ronak-59 @mahikaw please review

**Comment by iamareebjamal** [comment]
Um.. You could just set default visibility to GONE

**Comment by mejariamol** [comment]
But also had to handle after swipe refresh.

**Comment by iamareebjamal** [comment]
data binding should handle it automatically. When it is possible through data binding, there's no point in adding complexity in presenter

**Comment by mejariamol** [comment]
yes, but just knowing events size view can't decide whether to hide or show empty view

**Comment by mejariamol** [comment]
For example even for items size 0, on swipe refresh , empty view should hide and only progress bar should appear and on complete it should appear if items size 0. 
Hence its just like implementing progress bar. So I think better to be  implemented in presenter view

**Comment by iamareebjamal** [comment]
This is adding additional complexity to otherwise very basic idea. It's not actually bad to show empty view while loading because there is nothing present right now. Besides, if you add this interaction, you have to check it on errors too, that the empty layout is shown again or not. Plus, remember that we'll be moving the progress bar itself from presenter to data binding by adding a field in model, so this step is actually going backwards. Once we add that field in model for describing transient state, we'll just add a boolean check to see that if it is transient, hide the empty view and if it's not, display it. But now if we move logic in presenter, we'll have to move it back to layout some time ahead

**Comment by mejariamol** [comment]
@iamareebjamal 
>we'll be moving the progress bar itself from presenter to data binding by adding a field in model

Yes, thats why let keep empty view visibility logic in presenter for now. both shall be moved to data binding together.

One thing is clear that we need this behaviour - 
>Adds showEmptyView method in interface
Hides empty view on page start
Hides empty view on swipe refres

So I think lets keep this logic in presenter for now. Will create issue for handling progress bar and empty view visibility using databinding.

**Comment by iamareebjamal** [comment]
Then please test that it works in all cases

**Comment by mejariamol** [comment]
I have tested on device. Will add test in presenters by morning. :)

**Comment by mejariamol** [comment]
@iamareebjamal  added tests. please review.

File: app/src/test/java/org/fossasia/openevent/app/presenter/AttendeePresenterTest.java [hunk 1]
```diff
@@ -122,10 +125,12 @@ public void shouldLoadAttendeesSuccessfully() {
         attendeesPresenter.loadAttendees(false);
 
         inOrder.verify(attendeesView).showProgressBar(true);
+        inOrder.verify(attendeesView).showEmptyView(false);
```
Comment (iamareebjamal) [inline app/src/test/java/org/fossasia/openevent/app/presenter/AttendeePresenterTest.java]: Remove the false cases and add them as separate tests. These tests have become very self dependent. Each test should handle only one aspect of the view.

Will simplify them later as this is a low priority issue

Comment (iamareebjamal) [inline app/src/test/java/org/fossasia/openevent/app/presenter/AttendeePresenterTest.java]: And add a case where error happens while updating

**Comment by mejariamol** [comment]
@iamareebjamal done :+1:

**Comment by iamareebjamal** [comment]
@arp95 @rohanagarwal94 @Shailesh351 guys, please review

**Comment by mejariamol** [comment]
@iamareebjamal done :+1:

**Comment by mejariamol** [comment]
@arp95 @rohanagarwal94 @Shailesh351 guys, please review

**Pull request merged**